api/worker.go

```diff
@@ -8,6 +8,21 @@ import (
 	"go.sia.tech/core/types"
 )
 
+type AccountsLockHandlerRequest struct {
+	HostKey   types.PublicKey `json:"hostKey"`
```

Comment:
This is kind of related to https://github.com/SiaFoundation/renterd/pull/214#discussion_r1142170909

The reason why we have a `HostKey` is only for the edge case of an account not existing. The worker knows to deterministically derive accounts ids so it will just try to do random stuff like trying to spend from an account that might not exist yet or fund it. The `HostKey` is used to assign a host in that case.

Unfortunately the bus can't sanity check that the host matches the account because only the workers have access to the master key used to derive the accounts. 

One one hand this is a bit annoying but on the other hand it's quite flexible because a third party app can use the bus to create accounts for hosts which are derived in a different way than our workers.